v2/client/invoker.go

```diff
@@ -61,14 +64,22 @@ func (r *receiveInvoker) Invoke(ctx context.Context, m binding.Message, respFn p
 				cecontext.LoggerFrom(ctx).Error(fmt.Errorf("cloudevent validation failed on response event: %v, %w", verr, err))
 			}
 		}
-		if respFn != nil {
-			var rm binding.Message
-			if resp != nil {
-				rm = (*binding.EventMessage)(resp)
+
+		// protocol can manual ack by the result
+		if respFn == nil {
+			if result != nil {
```

Comment:
!= nil is not quite enough, result is a protocol.Result, and it has to be tested for a NACK directly